system/Images/Handlers/ImageMagickHandler.php

```diff
@@ -288,23 +288,34 @@ public function save(string $target = null, int $quality = 90): bool
 	{
 		$target = empty($target) ? $this->image() : $target;
 
-		// If no new resource has been created, then we're
-		// simply copy the existing one.
-		if (empty($this->resource))
+		// If no new resource has been created,  and we
+		// aren't adjusting the quality then we're simply
+		// copy the existing one.
+		if (empty($this->resource) && $quality == 100)
 		{
+			// we're moving it to the same spot so don't do anything.
+			if($target === null)
+			{
+				return true;
+			}
+
 			$name = basename($target);
 			$path = pathinfo($target, PATHINFO_DIRNAME);
 
 			return $this->image()->copy($path, $name);
 		}
 
+		$source = $this->image()->getPathname();
```

Comment:
What if we already made some operations on the image? Not using `$this->resource` means that we will lose everything.

Reply:
you're right, missed copying the resource over if it exists.